node/src/Pos/Block/Network/Logic.hs

```diff
@@ -348,6 +342,28 @@ handleRequestedHeaders cont inRecovery headers = do
             logDebug msg
             throwM $ DialogUnexpected msg
   where
+    newestHeader = headers ^. _NewestFirst . _neHead
+    oldestHeader = headers ^. _NewestFirst . _neLast
+    newestHash = headerHash newestHeader
+    oldestHash = headerHash oldestHeader
+    oldestEpoch = oldestHeader ^. epochIndexL
+
+    tryCalculateLrc = do
+        tipEpoch <- view epochIndexL <$> DB.getTipHeader @ssc
+        if oldestEpoch == tipEpoch + 1 then do
+            logDebug "handleRequesteHeaders: LRC started"
+            let handler UnknownBlocksForLrc =
+                    logWarning $
+                    "handleRequestedHeaders: tried lrcSingleShot, " <>
+                    "got UnknownBlocksForLrc"
+                handler e                   = throwM e
+            withStateLockNoMetrics LowPriority $ const $
+                lrcSingleShot oldestEpoch `catch` handler
+            logDebug "handleRequesteHeaders: LRC ended"
+        else logDebug $ sformat ("handleRequestHeaders: oldest header epoch ("%build%
+                                 ") /= tip epoch ("%build%" + 1)")
```

Comment:
This message looks a bit confusing. I mean `tip epoch (tipEpoch + 1)`.

Reply:
I've fixed both by amending to the last commit.